src/main/java/it/gov/pagopa/arc/service/AccessTokenBuilderService.java

```diff
@@ -0,0 +1,52 @@
+package it.gov.pagopa.arc.service;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import it.gov.pagopa.arc.config.JWTConfiguration;
+import it.gov.pagopa.arc.model.generated.TokenResponse;
+import it.gov.pagopa.arc.utils.CertUtils;
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.time.Instant;
+import java.util.UUID;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AccessTokenBuilderService {
+
+  private final String allowedAudience;
+  private final int expireIn;
+  private final RSAPublicKey rsaPublicKey;
+  private final RSAPrivateKey rsaPrivateKey;
+
+  public AccessTokenBuilderService(
+      JWTConfiguration jwtConfiguration
+  ) {
+    this.allowedAudience = jwtConfiguration.getAudience();
+    this.expireIn = jwtConfiguration.getAccessToken().getExpireIn();
+
+    try {
+      rsaPrivateKey = CertUtils.pemKey2PrivateKey(jwtConfiguration.getAccessToken().getPrivateKey());
+      rsaPublicKey = CertUtils.pemPub2PublicKey(jwtConfiguration.getAccessToken().getPublicKey());
+    } catch (InvalidKeySpecException | NoSuchAlgorithmException | IOException e) {
+      throw new IllegalStateException("Cannot load private and/or public key", e);
+    }
+  }
+
+  public TokenResponse build(){
```

Comment:
There is a violation between layers here. We are injecting a controller model inside the service. Try to isolate the layers and move the building of the response elsewhere

Reply:
fixed by returning String